src/cordova/info.js

```diff
@@ -17,93 +17,104 @@ specific language governing permissions and limitations
 
+/**
+ * Outputs information about your system, your Cordova installation and
+ * the Cordova project this is called in, if applicable.
+ *
+ * Useful when submitting bug reports and asking for help.
+ *
+ * @return {Promise} resolved when info has been dumped
+ */
 module.exports = function info () {
-    // Get projectRoot
-    var projectRoot = cordova_util.cdProjectRoot();
-    var output = '';
-    if (!projectRoot) {
-        return Q.reject(new Error('Current working directory is not a Cordova-based project.'));
-    }
+    const basicInfo = [
+        // Get versions for cordova and all direct cordova dependencies
+        cordovaVersionInfo(),
+        // Get information on the current environment
+        environmentInformation()
+    ];
 
-    // Array of functions, Q.allSettled
-    console.log('Collecting Data...\n\n');
-    return Q.allSettled([
-        // Get Node version
-        Q('Node version: ' + process.version),
-        // Get Cordova version
-        Q('Cordova version: ' + pkg.version),
-        // Get project config.xml file using ano
-        getProjectConfig(projectRoot),
+    const projectRoot = cordova_util.isCordova();
+    const projectInfo = !projectRoot ? [] : [
         // Get list of plugins
         listPlugins(projectRoot),
         // Get Platforms information
-        getPlatforms(projectRoot)
-    ]).then(function (promises) {
-        promises.forEach(function (p) {
-            output += p.state === 'fulfilled' ? p.value + '\n\n' : p.reason + '\n\n';
-        });
-        console.info(output);
-        fs.writeFile(path.join(projectRoot, 'info.txt'), output, 'utf-8', function (err) {
-            if (err) throw err;
-        });
-    });
+        getPlatforms(projectRoot),
+        // Display project config.xml file
+        displayFileContents(cordova_util.projectConfig(projectRoot)),
+        // Display project package.json file
+        displayFileContents(path.join(projectRoot, 'package.json'))
+    ];
+
+    const infoPromises = [].concat(basicInfo, projectInfo);
+    const failSafePromises = infoPromises.map(p => Q(p).catch(err => err));
+    return Q.all(failSafePromises)
+        .then(results => console.info(results.join('\n\n')));
 };
 
-function getPlatforms (projectRoot) {
-    var platforms = cordova_util.listPlatforms(projectRoot);
-    if (platforms.length) {
-        return Q.all(platforms.map(function (p) {
-            return getPlatformInfo(p, projectRoot);
-        })).then(function (outs) {
-            return outs.join('\n\n');
-        });
-    }
-    return Q.reject('No Platforms Currently Installed');
+function cordovaVersionInfo () {
+    const versionFor = name => require(`${name}/package`).version;
+    const deps = Object.keys(pkg.dependencies)
+        .filter(name => name.startsWith('cordova-'))
+        .map(name => `${name}@${versionFor(name)}`)
+        .join('\n');
+    return `Cordova version ${pkg.version} with:\n${indent(deps)}`;
```

Comment:
=> Cordova CLI version ...

Reply:
Oh. Ohhhh.

Well, that's too complicated for me ;)

How about outputting the CLI version in the "headline" and adding `cordova-lib` in the list of "packages" then? That would be what some user (read: me) expects and still have all the information you need.

Reply:
IIRC, that would be difficult with the current approach (we are looking at our dependencies, CLI depends on us though).

This should definitely go onto our TODO list though.

Reply:
Now an issue at: https://github.com/apache/cordova-lib/issues/659

Reply:
Change it to `Cordova lib version` here in the meantime to avoid confusion?

Reply:
:+1: